src/lib/base/buf_comp.h

```diff
@@ -101,6 +101,13 @@ class BOTAN_DLL Buffered_Computation
          return output;
          }
 
+      std::vector<byte> final_stdvec()
```

Comment:
Mostly because it's very unfortunate at runtime - many times a hash/MAC output is not secret and `unlock(final())` bounces an allocation+deallocation in the pool allocator for no reason. Many places in the code instead use something like

```
std::vector<byte> out(h->output_length())
h->final(out.data());
```

to avoid this, they could instead use the (definitely unfortunately named) `final_stdvec` instead. I'm fine with painting this bikeshed a different color.